x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/SSLService.java

```diff
@@ -656,18 +657,34 @@ public static Settings getHttpTransportSSLSettings(Settings settings) {
         return builder.build();
     }
 
-    private static List<Settings> getMonitoringExporterSettings(Settings settings) {
-        List<Settings> sslSettings = new ArrayList<>();
+    public SSLConfiguration getHttpTransportSSLConfiguration() {
+        return getSSLConfiguration(XPackSettings.HTTP_SSL_PREFIX);
+    }
+
+    private static Map<String, Settings> getMonitoringExporterSettings(Settings settings) {
+        Map<String, Settings> sslSettings = new HashMap<>();
         Map<String, Settings> exportersSettings = settings.getGroups("xpack.monitoring.exporters.");
         for (Entry<String, Settings> entry : exportersSettings.entrySet()) {
             Settings exporterSSLSettings = entry.getValue().getByPrefix("ssl.");
             if (exporterSSLSettings.isEmpty() == false) {
-                sslSettings.add(exporterSSLSettings);
+                sslSettings.put("xpack.monitoring.exporters." + entry.getKey() + ".ssl", exporterSSLSettings);
             }
         }
         return sslSettings;
     }
 
+    public SSLConfiguration getSSLConfiguration(String contextName) {
+        if (contextName.endsWith(".")) {
+            contextName = contextName.substring(0, contextName.length() - 1);
+        }
+        final SSLConfiguration configuration = sslConfigurations.get(contextName);
+        if (configuration == null && logger.isTraceEnabled()) {
+            logger.trace("Cannot find SSL configuration for context {}. Known contexts are: {}", contextName,
```

Comment:
hmm this feels more like a warn message? This should only happen if there is a bug?

Reply:
It used to be possible/common for monitoring, due to dynamic monitoring SSL settings - there was no guarantee that the "context name" was known when the SSLService was loaded.

Now that has changed so that monitoring only calls this method if it knows it is using secure-settings, so a missing configuration is a bug (since a config with secure settings needs to exist at node startup)